### ISSUE TITLE
threadMXBeanTestSuite1_7_FAILED Segmentation error vmState=0x00051dff

### ISSUE THREAD

**Issue by JasonFengJ9** [eclipse-openj9/openj9#19197]
Failure link
------------

From [an internal build](https://hyc-runtimes-jenkins.swg-devops.com/job/Test_openjdk11_j9_extended.functional_x86-64_windows_testList_2/487/)(`win19x86-rtp-rt7-1`):
```
java version "11.0.23-beta" 2024-04-16
IBM Semeru Runtime Certified Edition 11.0.23+6-202403192217 (build 11.0.23-beta+6-202403192217)
Eclipse OpenJ9 VM 11.0.23+6-202403192217 (build master-6f741bc32, JRE 11 Windows Server 2019 amd64-64-Bit Compressed References 20240319_796 (JIT enabled, AOT enabled)
OpenJ9   - 6f741bc32
OMR      - 1bf2ef421
JCL      - 00819f7d20 based on jdk-11.0.23+6)
```
[Rerun in Grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/parambuild/?SDK_RESOURCE=customized&amp;TARGET=threadMXBeanTestSuite1_7&amp;BASE_DOCKER_REGISTRY_CREDENTIAL_ID=&amp;TEST_FLAG=&amp;UPSTREAM_TEST_JOB_NAME=Test_openjdk11_j9_extended.functional_x86-64_windows&amp;DOCKER_REQUIRED=false&amp;ACTIVE_NODE_TIMEOUT=5&amp;VENDOR_TEST_DIRS=functional&amp;EXTRA_DOCKER_ARGS=&amp;TKG_OWNER_BRANCH=adoptium%3Amaster&amp;OPENJ9_SYSTEMTEST_OWNER_BRANCH=eclipse%3Amaster&amp;PLATFORM=x86-64_windows&amp;GENERATE_JOBS=true&amp;KEEP_REPORTDIR=false&amp;PERSONAL_BUILD=false&amp;DOCKER_REGISTRY_DIR=&amp;RERUN_ITERATIONS=0&amp;ADOPTOPENJDK_REPO=https%3A%2F%2Fgithub.com%2Fadoptium%2Faqa-tests.git&amp;SETUP_JCK_RUN=false&amp;DOCKER_REGISTRY_URL_CREDENTIAL_ID=&amp;LABEL=&amp;EXTRA_OPTIONS=&amp;CUSTOMIZED_SDK_URL=https%3A%2F%2Fna-public.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk11u%2Fjdk11u-windows-x64-openj9-IBM%2F796%2Fibm-semeru-certified-jdk_x64_windows_JDK11U_2024-03-19-22-00.zip+https%3A%2F%2Fna-public.artifactory.swg-devops.com%2Fartifactory%2Fsys-rt-generic-local%2Fhyc-runtimes-jenkins.swg-devops.com%2Fbuild-scripts%2Fjobs%2Fjdk11u%2Fjdk11u-windows-x64-openj9-IBM%2F796%2Fibm-semeru-certified-testimage_x64_windows_JDK11U_2024-03-19-22-00.zip&amp;BUILD_IDENTIFIER=&amp;JENKINS_KEY=Jenkins+4096+key&amp;ADOPTOPENJDK_BRANCH=master&amp;LIGHT_WEIGHT_CHECKOUT=false&amp;USE_JRE=false&amp;ARTIFACTORY_SERVER=na.artifactory.swg-devops&amp;KEEP_WORKSPACE=false&amp;USER_CREDENTIALS_ID=83181e25-eea4-4f55-8b3e-e79615733226&amp;JDK_VERSION=11&amp;DOCKER_REGISTRY_URL=&amp;ITERATIONS=1&amp;VENDOR_TEST_REPOS=git%40github.ibm.com%3Aruntimes%2Ftest.git&amp;JDK_REPO=git%40github.com%3Aibmruntimes%2Fopenj9-openjdk-jdk11&amp;JCK_GIT_BRANCH=main&amp;RELEASE_TAG=&amp;OPENJ9_BRANCH=master&amp;OPENJ9_SHA=&amp;JCK_GIT_REPO=&amp;VENDOR_TEST_BRANCHES=master&amp;OPENJ9_REPO=https%3A%2F%2Fgithub.com%2Feclipse-openj9%2Fopenj9.git&amp;UPSTREAM_JOB_NAME=&amp;CLOUD_PROVIDER=&amp;CUSTOM_TARGET=&amp;VENDOR_TEST_SHAS=&amp;JDK_BRANCH=openj9&amp;LABEL_ADDITION=&amp;ARTIFACTORY_REPO=sys-rt-generic-local%2Chyc-runtimes-team-restricted-oracle-sdk-generic-local&amp;ARTIFACTORY_ROOT_DIR=&amp;UPSTREAM_TEST_JOB_NUMBER=528&amp;DOCKERIMAGE_TAG=&amp;JDK_IMPL=openj9&amp;TEST_TIME=&amp;SSH_AGENT_CREDENTIAL=83181e25-eea4-4f55-8b3e-e79615733226&amp;AUTO_DETECT=true&amp;SLACK_CHANNEL=%23rt-jenkins&amp;DYNAMIC_COMPILE=true&amp;RELATED_NODES=&amp;ADOPTOPENJDK_SYSTEMTEST_OWNER_BRANCH=adoptium%3Amaster&amp;APPLICATION_OPTIONS=&amp;CUSTOMIZED_SDK_URL_CREDENTIAL_ID=7c1c2c28-650f-49e0-afd1-ca6b60479546&amp;ARCHIVE_TEST_RESULTS=false&amp;NUM_MACHINES=&amp;OPENJDK_SHA=&amp;TRSS_URL=http%3A%2F%2Ftrssrtp1.fyre.ibm.com%2F&amp;USE_TESTENV_PROPERTIES=false&amp;BUILD_LIST=functional&amp;ADDITIONAL_ARTIFACTS_REQUIRED=&amp;UPSTREAM_JOB_NUMBER=&amp;STF_OWNER_BRANCH=adoptium%3Amaster&amp;TIME_LIMIT=25&amp;JVM_OPTIONS=&amp;PARALLEL=None) - Change TARGET to run only the failed test targets.

Optional info
-------------

Failure output (captured from console output)
---------------------------------------------
```
[2024-03-20T04:54:41.079Z] variation: Mode610 -Xjit:noJitUntilMain,count=0,optlevel=warm
[2024-03-20T04:54:41.079Z] JVM_OPTIONS:  -Xcompressedrefs -Xjit -Xgcpolicy:gencon -Xjit:noJitUntilMain,count=0,optlevel=warm 

[2024-03-20T04:54:43.376Z] TESTING:
[2024-03-20T04:55:24.412Z] [IncludeExcludeTestAnnotationTransformer] [INFO] EXCLUDE_FILE environment variable: C:/Users/jenkins/workspace/Test_openjdk11_j9_extended.functional_x86-64_windows_testList_2/aqa-tests/\\TestConfig\\resources\\excludes\\latest_exclude_11.txt
[2024-03-20T04:55:24.412Z] [IncludeExcludeTestAnnotationTransformer] [INFO] Processing exclude file: C:/Users/jenkins/workspace/Test_openjdk11_j9_extended.functional_x86-64_windows_testList_2/aqa-tests/\\TestConfig\\resources\\excludes\\latest_exclude_11.txt
[2024-03-20T04:55:24.412Z] ...
[2024-03-20T04:55:24.412Z] ... TestNG 6.14.2 by Cédric Beust (cedric@beust.com)
[2024-03-20T04:55:24.412Z] ...
[2024-03-20T04:55:24.412Z] 
[2024-03-20T04:55:26.203Z] Unhandled exception
[2024-03-20T04:55:26.203Z] Type=Segmentation error vmState=0x00051dff
[2024-03-20T04:55:26.203Z] Windows_ExceptionCode=c0000005 J9Generic_Signal=00000004 ExceptionAddress=00007FFC5441CCEF ContextFlags=0010005f
[2024-03-20T04:55:26.203Z] Handler1=00007FFC5445EB70 Handler2=00007FFC58BFAC50 InaccessibleReadAddress=0000008A0000001A
[2024-03-20T04:55:26.203Z] RDI=0000008A86AF9740 RSI=00007FF4190C2FB0 RAX=0000000000000000 RBX=0000008A86AF9870
[2024-03-20T04:55:26.203Z] RCX=0000008A86AF96E8 RDX=0000008A86AF96E0 R8=0000008A00000001 R9=00007FF4190E0260
[2024-03-20T04:55:26.203Z] R10=0000000000000007 R11=0000000000000001 R12=00007FF4190B0660 R13=0000000000000000
[2024-03-20T04:55:26.203Z] R14=0000000000000054 R15=0000000000000001
[2024-03-20T04:55:26.203Z] RIP=00007FFC5441CCEF RSP=0000008A86AF9640 RBP=0000008A86AF9770 EFLAGS=0000000000010202
[2024-03-20T04:55:26.203Z] FS=0053 ES=002B DS=002B
[2024-03-20T04:55:26.203Z] XMM0 0000008a86af9840 (f: 2259654656.000000, d: 2.939518e-312)
[2024-03-20T04:55:26.203Z] XMM1 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2024-03-20T04:55:26.203Z] XMM2 4564657470757272 (f: 1886745216.000000, d: 1.972610e+26)
[2024-03-20T04:55:26.203Z] XMM3 00007ff419180b78 (f: 421006208.000000, d: 6.950830e-310)
[2024-03-20T04:55:26.203Z] XMM4 00007ff419180db8 (f: 421006784.000000, d: 6.950830e-310)
[2024-03-20T04:55:26.203Z] XMM5 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2024-03-20T04:55:26.203Z] XMM6 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2024-03-20T04:55:26.203Z] XMM7 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2024-03-20T04:55:26.203Z] XMM8 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2024-03-20T04:55:26.203Z] XMM9 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2024-03-20T04:55:26.203Z] XMM10 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2024-03-20T04:55:26.203Z] XMM11 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2024-03-20T04:55:26.203Z] XMM12 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2024-03-20T04:55:26.203Z] XMM13 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2024-03-20T04:55:26.203Z] XMM14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2024-03-20T04:55:26.203Z] XMM15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
[2024-03-20T04:55:26.203Z] Module=C:\Users\jenkins\workspace\Test_openjdk11_j9_extended.functional_x86-64_windows_testList_2\jdkbinary\j2sdk-image\bin\default\j9jit29.dll
[2024-03-20T04:55:26.203Z] Module_base_address=00007FFC543F0000 Offset_in_DLL=000000000002ccef
[2024-03-20T04:55:26.203Z] 
[2024-03-20T04:55:26.203Z] Method_being_compiled=org/openj9/test/java/lang/management/ThreadMXBean/FindDeadlockTest$DSThread.run()V
[2024-03-20T04:55:26.203Z] Target=2_90_20240319_796 (Windows Server 2019 10.0 build 17763)
[2024-03-20T04:55:26.203Z] CPU=amd64 (4 logical CPUs) (0x1fff78000 RAM)
[2024-03-20T04:55:26.203Z] ----------- Stack Backtrace -----------
[2024-03-20T04:55:26.204Z] (0x00007FFC5441CCEF [j9jit29+0x2ccef])
[2024-03-20T04:55:26.204Z] Java_java_lang_invoke_ThunkTuple_initialInvokeExactThunk+0x2bd7c3 (0x00007FFC547797A3 [j9jit29+0x3897a3])
[2024-03-20T04:55:26.204Z] Java_java_lang_invoke_ThunkTuple_initialInvokeExactThunk+0x2bb4dc (0x00007FFC547774BC [j9jit29+0x3874bc])
[2024-03-20T04:55:26.204Z] Java_java_lang_invoke_ThunkTuple_initialInvokeExactThunk+0x3ae3f0 (0x00007FFC5486A3D0 [j9jit29+0x47a3d0])
[2024-03-20T04:55:26.204Z] Java_java_lang_invoke_ThunkTuple_initialInvokeExactThunk+0x3aae65 (0x00007FFC54866E45 [j9jit29+0x476e45])
[2024-03-20T04:55:26.204Z] Java_java_lang_invoke_ThunkTuple_initialInvokeExactThunk+0x24904f (0x00007FFC5470502F [j9jit29+0x31502f])
[2024-03-20T04:55:26.204Z] (0x00007FFC5444C2A4 [j9jit29+0x5c2a4])
[2024-03-20T04:55:26.204Z] (0x00007FFC5444F885 [j9jit29+0x5f885])
[2024-03-20T04:55:26.204Z] j9port_isCompatible+0x18b06 (0x00007FFC58BFB6E6 [j9prt29+0x1b6e6])
[2024-03-20T04:55:26.204Z] j9port_isCompatible+0x1a240 (0x00007FFC58BFCE20 [j9prt29+0x1ce20])
[2024-03-20T04:55:26.204Z] (0x00007FFC5444BA13 [j9jit29+0x5ba13])
[2024-03-20T04:55:26.204Z] (0x00007FFC54451D35 [j9jit29+0x61d35])
[2024-03-20T04:55:26.204Z] (0x00007FFC5445171A [j9jit29+0x6171a])
[2024-03-20T04:55:26.204Z] (0x00007FFC5445E9A0 [j9jit29+0x6e9a0])
[2024-03-20T04:55:26.204Z] j9port_isCompatible+0x1a27b (0x00007FFC58BFCE5B [j9prt29+0x1ce5b])
[2024-03-20T04:55:26.204Z] (0x00007FFC5445E696 [j9jit29+0x6e696])
[2024-03-20T04:55:26.204Z] omrthread_get_category+0xa32 (0x00007FFC5B044202 [j9thr29+0x4202])
[2024-03-20T04:55:26.204Z] _o_exp+0x5a (0x00007FFC67F6268A [ucrtbase+0x2268a])
[2024-03-20T04:55:26.204Z] BaseThreadInitThunk+0x14 (0x00007FFC6B0B7AC4 [KERNEL32+0x17ac4])
[2024-03-20T04:55:26.204Z] RtlUserThreadStart+0x21 (0x00007FFC6B73A4E1 [ntdll+0x5a4e1])
[2024-03-20T04:55:26.204Z] ---------------------------------------
[2024-03-20T04:55:26.204Z] JVMDUMP039I Processing dump event "gpf", detail "" at 2024/03/19 21:55:25 - please wait.

[2024-03-20T04:55:39.518Z] [FindDeadlockTest] [ERROR] TEST SETUP FAILED
[2024-03-20T04:55:39.518Z] org.openj9.test.java.lang.management.ThreadMXBean.FindDeadlockTest$TestSetupFailedException: Thread[ds3,5,main] did not block.
[2024-03-20T04:55:39.518Z] 	at org.openj9.test.java.lang.management.ThreadMXBean.FindDeadlockTest.checkedGetBlocker(FindDeadlockTest.java:427)
[2024-03-20T04:55:39.518Z] 	at org.openj9.test.java.lang.management.ThreadMXBean.FindDeadlockTest.startDeadlockedSynchronizers(FindDeadlockTest.java:260)
[2024-03-20T04:55:39.518Z] 	at org.openj9.test.java.lang.management.ThreadMXBean.FindDeadlockTest.testFindDeadlockTest(FindDeadlockTest.java:52)
[2024-03-20T04:55:39.518Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[2024-03-20T04:55:39.518Z] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[2024-03-20T04:55:39.518Z] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[2024-03-20T04:55:39.518Z] 	at java.base/java.lang.reflect.Method.invoke(Method.java:572)
[2024-03-20T04:55:39.518Z] 	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:124)
[2024-03-20T04:55:39.518Z] 	at org.testng.internal.Invoker.invokeMethod(Invoker.java:580)
[2024-03-20T04:55:39.518Z] 	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:716)
[2024-03-20T04:55:39.518Z] 	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:988)
[2024-03-20T04:55:39.518Z] 	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:125)
[2024-03-20T04:55:39.518Z] 	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:109)
[2024-03-20T04:55:39.518Z] 	at org.testng.TestRunner.privateRun(TestRunner.java:648)
[2024-03-20T04:55:39.518Z] 	at org.testng.TestRunner.run(TestRunner.java:505)
[2024-03-20T04:55:39.518Z] 	at org.testng.SuiteRunner.runTest(SuiteRunner.java:455)
[2024-03-20T04:55:39.518Z] 	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:450)
[2024-03-20T04:55:39.518Z] 	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:415)
[2024-03-20T04:55:39.518Z] 	at org.testng.SuiteRunner.run(SuiteRunner.java:364)
[2024-03-20T04:55:39.518Z] 	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
[2024-03-20T04:55:39.518Z] 	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:84)
[2024-03-20T04:55:39.518Z] 	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1208)
[2024-03-20T04:55:39.518Z] 	at org.testng.TestNG.runSuitesLocally(TestNG.java:1137)
[2024-03-20T04:55:39.518Z] 	at org.testng.TestNG.runSuites(TestNG.java:1049)
[2024-03-20T04:55:39.518Z] 	at org.testng.TestNG.run(TestNG.java:1017)
[2024-03-20T04:55:39.518Z] 	at org.testng.TestNG.privateMain(TestNG.java:1354)
[2024-03-20T04:55:39.518Z] 	at org.testng.TestNG.main(TestNG.java:1323)
[2024-03-20T04:55:39.518Z] Caused by: org.openj9.test.java.lang.management.ThreadMXBean.FindDeadlockTest$TestSetupFailedException: Thread[ds3,5,main] stack trace:
[2024-03-20T04:55:39.879Z] JVMDUMP010I JIT dump written to C:\Users\jenkins\workspace\Test_openjdk11_j9_extended.functional_x86-64_windows_testList_2\aqa-tests\TKG\output_17109047583321\threadMXBeanTestSuite1_7\jitdump.20240319.215525.10796.0004.dmp
[2024-03-20T04:55:39.879Z] JVMDUMP013I Processed dump event "gpf", detail "".
[2024-03-20T04:55:39.879Z] -----------------------------------
[2024-03-20T04:55:39.879Z] threadMXBeanTestSuite1_7_FAILED
```

[50x internal grinder](https://hyc-runtimes-jenkins.swg-devops.com/job/Grinder/38966/) - https://github.com/eclipse-openj9/openj9/issues/19197#issuecomment-2011073517

**Comment by pshipton** [comment]
`vmState [0x51dff]: {J9VMSTATE_JIT} {deadTreesElimination}`

@hzongaro fyi

**Comment by pshipton** [comment]
Grinder results
Failed 7/10 on win11x86-svl-rt8-1
Passed on win11x86-rtp-rt3-1
Failed 6/10 on win10x64vs6
Failed 4/10 on win16x64rt2-8
Failed 6/10 on win19x86-svl-rt7-1

**Comment by JamesKingdon** [comment]
This looks reminiscent of a series of cases I've been working on where we appear to have corrupted 64 bit pointers with small int values in the low 32 bits.
InaccessibleReadAddress=0000008A0000001A
R8=0000008A00000001

**Comment by hzongaro** [comment]
It looks like it's crashing while walking through `longestPaths` at the end of [`TR::DeadTreesElimination::process`](https://github.com/eclipse/omr/blob/1bf2ef421caa77bd678293575dc6ff3e1e173b09/compiler/optimizer/DeadTreesElimination.cpp#L759-L762), deallocating its contents.  It crashes in `std::_Tree_val::_Erase_tree` at 0x7FFC5441CCEF called from 0x7FFC5477979E:

```
    j9jit29!std::_Tree_val<std::_Tree_simple_types<std::pair<int const ,TR_Stack<TR::SymbolReference *> *> > >::_Erase_tree<TR::typed_allocator<std::_Tree_node<std::pair<int const ,TR_Stack<TR::SymbolReference *> *>,void *>,TR::Region &> >:
00007ffc`5441cce0 48895c2408     mov     qword ptr [rsp+8], rbx
00007ffc`5441cce5 4889742410     mov     qword ptr [rsp+10h], rsi
00007ffc`5441ccea 57             push    rdi
00007ffc`5441cceb 4883ec20       sub     rsp, 20h
00007ffc`5441ccef 4180781900     cmp     byte ptr [_Rootnode->_Isnil (r8+19h)], 0  <<<<< Crash here
```

from end of `TR::DeadTreesElimination::process`:

```
00007ffc`5477978a 7538               jne     j9jit29!TR::DeadTreesElimination::process+0x11a4 (7ffc547797c4)
00007ffc`5477978c 0f1f4000           nop     dword ptr [this{->_optionSets(!!)} (rax)]
00007ffc`54779790 4c8b4310           mov     r8, qword ptr [_Rootnode->_Right (rbx+10h)]
00007ffc`54779794 488d542470         lea     longestPaths (rdx), [longestPaths{._Mypair._Myval2._Myval1._backingAllocator} (rsp+70h)]
00007ffc`54779799 488d4c2478         lea     rcx, [longestPaths._Mypair._Myval2._Myval2{._Myhead} (rsp+78h)]
00007ffc`5477979e e83d35caff         call    j9jit29!std::_Tree_val<std::_Tree_simple_types<std::pair<int const , TR_Stack<TR::SymbolReference *> *> > >::_Erase_tree<TR::typed_allocator<std::_Tree_node<std::pair<int const , TR_Stack<TR::SymbolReference *> *>, void *>, TR::Region &> > (7ffc5441cce0)
00007ffc`547797a3 488bd3             mov     rdx, _Rootnode (rbx)
00007ffc`547797a6 488b1b             mov     _Rootnode (rbx), qword ptr [_Rootnode (rbx)]
00007ffc`547797a9 41b830000000       mov     r8d, 30h
00007ffc`547797af 488b4c2470         mov     rcx, qword ptr [longestPaths{._Mypair._Myval2._Myval1._backingAllocator} (rsp+70h)]
00007ffc`547797b4 e86782f9ff         call    j9jit29!TR::Region::deallocate (7ffc54711a20)
00007ffc`547797b9 807b1900           cmp     byte ptr [_Rootnode->_Isnil (rbx+19h)], 0
00007ffc`547797bd 74d1               je      j9jit29!TR::DeadTreesElimination::process+0x1170 (7ffc54779790)
```

I'm trying to add a small debugging method that will walk through `longestPaths` periodically throughout `DeadTreesElimination::process` if tracing is enabled in hopes of narrowing down where it gets corrupted.  Hopefully that doesn't end up masking the problem.

**Comment by hzongaro** [comment]
I think I have figured out what is happening.  In [`OMR::DeadTreesElimination::process(TR::TreeTop*,TR::TreeTop*)`](https://github.com/eclipse/omr/blob/8037ccd0406c017f6a8043a8d04e19ac68f626f2/compiler/optimizer/DeadTreesElimination.cpp#L759-L766), a `TR::StackMemoryRegion` object is created, and it is used to allocate objects named `longestPaths` and `anchors`.  The `process` method calls [`isSafeToReplaceNode`](https://github.com/eclipse/omr/blob/8037ccd0406c017f6a8043a8d04e19ac68f626f2/compiler/optimizer/DeadTreesElimination.cpp#L890-L899), passing `_targetTrees`, which is a `List<OMR::TreeInfo>`, as an argument.  In turn, that function calls [`findOrCreateTreeInfo`](https://github.com/eclipse/omr/blob/8037ccd0406c017f6a8043a8d04e19ac68f626f2/compiler/optimizer/DeadTreesElimination.cpp#L66-L79), passing the `_targetTrees` object along.

`findOrCreateTreeInfo` will look for an `OMR::TreeInfo` object in `_targetTrees` that refers to the current `TR::TreeTop *`, or it will create a new `OMR::TreeInfo` object in stack memory.  However, `_targetTrees` is initialized in the [constructor of `DeadTreesElimination`](https://github.com/eclipse/omr/blob/8037ccd0406c017f6a8043a8d04e19ac68f626f2/compiler/optimizer/DeadTreesElimination.cpp#L435-L441), and its lifetime extends past the call to `OMR::DeadTreesElimination::process(TR::TreeTop*,TR::TreeTop*)`.  Thus `OMR::TreeInfo` instances in the `List` might have been allocated from stack memory that has already been released, and whose storage is now being reused by other data structures, the aforementioned  `longestPaths` and `anchors`.  The `anchors` structure contains `TR::TreeTop` pointers, so if such a pointer is stored into memory that's shared with the `_treeTop` field of an `OMR::TreeInfo` that was allocated from that now reused stack memory, `findOrCreateTreeInfo` will incorrectly return that `OMR::TreeInfo` object for a `TR::TreeTop *` that happens to match it.  Then a call to set the `_height` field of that `OMR::TreeInfo` will corrupt whatever data was at that position in the `anchors` structure that happened to share memory with the now obsolete `OMR::TreeInfo`, ultimately leading to a crash.

I believe the fix is to allocate the `OMR::TreeInfo` objects using the `TR::Region` that is associated with `_targetTrees`.  I am testing that fix.

**Comment by pshipton** [comment]
This also occurs in the 0.44 release.
https://openj9-jenkins.osuosl.org/job/Test_openjdk11_j9_extended.functional_x86-64_windows_Release_testList_1/91 - [win2019x64-openj9-3a](https://openj9-jenkins.osuosl.org/computer/win2019x64-openj9-3a)
threadMXBeanTestSuite1_7

https://openj9-artifactory.osuosl.org/artifactory/ci-openj9/Test/Test_openjdk11_j9_extended.functional_x86-64_windows_Release_testList_1/91/functional_test_output.tar.gz
```
17:59:36  Type=Segmentation error vmState=0x00051dff
17:59:36  Windows_ExceptionCode=c0000005 J9Generic_Signal=00000004 ExceptionAddress=00007FFB3FDFCCDF ContextFlags=0010005f
17:59:36  Handler1=00007FFB3FE3E550 Handler2=00007FFB43ECABA0 InaccessibleReadAddress=0000005E0000001A
17:59:36  RDI=0000005EF4AF90D0 RSI=00007FF4F4E42FB0 RAX=0000000000000000 RBX=0000005EF4AF9200
17:59:36  RCX=0000005EF4AF9070 RDX=0000005EF4AF9068 R8=0000005E00000001 R9=00007FF4F4E60260
17:59:36  R10=0000000000000007 R11=0000000000000001 R12=0000000000000000 R13=00007FF4F4E439B0
17:59:36  R14=0000000000000054 R15=0000000000000001
17:59:36  RIP=00007FFB3FDFCCDF RSP=0000005EF4AF8FC0 RBP=0000005EF4AF90F0 EFLAGS=0000000000010206
17:59:36  FS=0053 ES=002B DS=002B
17:59:36  XMM0 0000005ef4af91d0 (f: 4105146880.000000, d: 2.014958e-312)
17:59:36  XMM1 0000000000000000 (f: 0.000000, d: 0.000000e+00)
17:59:36  XMM2 4564657470757272 (f: 1886745216.000000, d: 1.972610e+26)
17:59:36  XMM3 00007ff4f4f00b78 (f: 4109372160.000000, d: 6.951012e-310)
17:59:36  XMM4 00007ff4f4f00db8 (f: 4109372928.000000, d: 6.951012e-310)
17:59:36  XMM5 0000000000000000 (f: 0.000000, d: 0.000000e+00)
17:59:36  XMM6 0000000000000000 (f: 0.000000, d: 0.000000e+00)
17:59:36  XMM7 0000000000000000 (f: 0.000000, d: 0.000000e+00)
17:59:36  XMM8 0000000000000000 (f: 0.000000, d: 0.000000e+00)
17:59:36  XMM9 0000000000000000 (f: 0.000000, d: 0.000000e+00)
17:59:36  XMM10 0000000000000000 (f: 0.000000, d: 0.000000e+00)
17:59:36  XMM11 0000000000000000 (f: 0.000000, d: 0.000000e+00)
17:59:36  XMM12 0000000000000000 (f: 0.000000, d: 0.000000e+00)
17:59:36  XMM13 0000000000000000 (f: 0.000000, d: 0.000000e+00)
17:59:36  XMM14 0000000000000000 (f: 0.000000, d: 0.000000e+00)
17:59:36  XMM15 0000000000000000 (f: 0.000000, d: 0.000000e+00)
17:59:36  Module=C:\Users\jenkins\workspace\Test_openjdk11_j9_extended.functional_x86-64_windows_Release_testList_1\jdkbinary\j2sdk-image\bin\default\j9jit29.dll
17:59:36  Module_base_address=00007FFB3FDD0000 Offset_in_DLL=000000000002ccdf
17:59:36  
17:59:36  Method_being_compiled=org/openj9/test/java/lang/management/ThreadMXBean/FindDeadlockTest$DSThread.run()V
17:59:36  Target=2_90_20240413_132 (Windows Server 2019 10.0 build 17763)
17:59:36  CPU=amd64 (4 logical CPUs) (0x3fff77000 RAM)
17:59:36  ----------- Stack Backtrace -----------
17:59:36  (0x00007FFB3FDFCCDF [j9jit29+0x2ccdf])
17:59:36  Java_java_lang_invoke_ThunkTuple_initialInvokeExactThunk+0x2bdeb2 (0x00007FFB40159072 [j9jit29+0x389072])
17:59:36  Java_java_lang_invoke_ThunkTuple_initialInvokeExactThunk+0x2bbafc (0x00007FFB40156CBC [j9jit29+0x386cbc])
17:59:36  Java_java_lang_invoke_ThunkTuple_initialInvokeExactThunk+0x3b0530 (0x00007FFB4024B6F0 [j9jit29+0x47b6f0])
17:59:36  Java_java_lang_invoke_ThunkTuple_initialInvokeExactThunk+0x3acf75 (0x00007FFB40248135 [j9jit29+0x478135])
17:59:36  Java_java_lang_invoke_ThunkTuple_initialInvokeExactThunk+0x2494e7 (0x00007FFB400E46A7 [j9jit29+0x3146a7])
17:59:36  (0x00007FFB3FE2BE0A [j9jit29+0x5be0a])
17:59:36  (0x00007FFB3FE2F3F8 [j9jit29+0x5f3f8])
17:59:36  j9port_isCompatible+0x18a46 (0x00007FFB43ECB626 [j9prt29+0x1b626])
17:59:36  j9port_isCompatible+0x1a180 (0x00007FFB43ECCD60 [j9prt29+0x1cd60])
17:59:36  (0x00007FFB3FE2B56E [j9jit29+0x5b56e])
17:59:36  (0x00007FFB3FE318A0 [j9jit29+0x618a0])
17:59:36  (0x00007FFB3FE3128A [j9jit29+0x6128a])
17:59:36  (0x00007FFB3FE3E380 [j9jit29+0x6e380])
17:59:36  j9port_isCompatible+0x1a1bb (0x00007FFB43ECCD9B [j9prt29+0x1cd9b])
17:59:36  (0x00007FFB3FE3E076 [j9jit29+0x6e076])
17:59:36  omrthread_get_category+0xa42 (0x00007FFB48484242 [j9thr29+0x4242])
17:59:36  _o_exp+0x5a (0x00007FFB527E268A [ucrtbase+0x2268a])
17:59:36  BaseThreadInitThunk+0x14 (0x00007FFB54B67AC4 [KERNEL32+0x17ac4])
17:59:36  RtlUserThreadStart+0x21 (0x00007FFB557FA4E1 [ntdll+0x5a4e1])
```

**Comment by hzongaro** [comment]
> This also occurs in the 0.44 release.

It's not a new problem - I believe it was first introduced in the 0.9.0 release, but it's highly intermittent.

**Comment by hzongaro** [comment]
@pshipton, @JamesKingdon, do you feel we should port the fix for this into the 0.44 release as well, or is it safe to wait for the 0.46 release?

**Comment by pshipton** [comment]
This isn't a new problem, regression, or a big customer impact? We should wait for 0.46. We've already done the Release Candidate 1 builds for 0.44. Adding anything is going to delay the release.

**Comment by JamesKingdon** [comment]
Does it make a difference to when it would get into IBM sdk?

**Comment by pshipton** [comment]
The fix is not in 24_02, or 24_02u1, it will be in 24_03. It could be double delivered to 24_02u1, there is time.

**Comment by hzongaro** [comment]
Fixed by eclipse/omr#7305
